lib/internal/url.js

```diff
@@ -932,6 +935,64 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  for (var n = 0; n < pathname.length; n++) {
+    if (pathname[n] === '%') {
+      var third = pathname.codePointAt(n + 2) | 0x20;
+      if ((pathname[n + 1] === '2' && third === 102) || // 2f 2F /
+          (pathname[n + 1] === '5' && third === 99)) {  // 5c 5C \
+        return new Error('Path must not include encoded \\ or / characters');
+      }
+    }
+  }
+  pathname = decodeURIComponent(pathname);
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
+    // Pass the hostname through domainToUnicode just in case
+    // it has been punycode encoded. We do not need to worry
+    // about percent encoding because the URL parser will have
+    // already taken care of that for us.
+    return `//${domainToUnicode(hostname)}${pathname}`;
```

Comment:
Okay, so `url.domainToUnicode()` only acts on (sub)domains that start with xn--.  Can you edit the comment to make clear the distinction between punycode decoding and IDN?  It led me to believe that `domainToUnicode(...)` is just `punycode.decode(...)`.